mm2src/coins/lp_coins.rs

```diff
@@ -1502,6 +1544,31 @@ pub struct SendMakerPaymentArgs<'a, Coin: CoinAssocTypes + ?Sized> {
     pub swap_unique_data: &'a [u8],
 }
 
+pub struct SendNftMakerPaymentArgs<'a, Coin: CoinAssocTypes + NftAssocTypes + ?Sized> {
```

Comment:
Again, we shouldn't have maker payment args specific for NFTs. It will create a lot of conditional code in swaps code, you can leave it for now for the P.O.C. purpose but we need to have a plan on how we will integrate NFT swaps logic in current swap state machine.

Reply:
Yep, its just one of two ways of two nft implementations from this https://github.com/KomodoPlatform/komodo-defi-framework/pull/2084#discussion_r1524514860 description

Reply:
> but we need to have a plan on how we will integrate NFT swaps logic in current swap state machine.

Oke, then we can think about approach with concatenated string of `"token_address,token_id"` parameter in Maker Payment args. If we want to reutulize the existing code, then we can add `is_nft:bool` param and optional `nft_key:&'a [u8]`. 
If `is_nft:true`, we try to parse string from `nft_key` and try to get `NftInfo` value from `nfts_infos` in global NFT. Well this means we will have to handle nft behavior and coin in same `States`,

Reply:
> If is_nft:true, nft_key must be Some, and we try to get NftInfo value from nfts_infos in global NFT. Well this means we will have to handle nft behavior and coin in same States.

Answered here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2084#discussion_r1524760947 , we will re-assess in next PRs I guess.